File: DSAA/week1/p1.java
import java.util.Scanner;

public class p1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = sc.nextInt();
        }
        int T = sc.nextInt();
        for (int i = 0; i < T; i++) {
            int num = sc.nextInt();
            boolean flag = false;
            for (int j = 0; j < n; j++) {
                if (A[j] == num) {
                    System.out.println("yes");
                    flag = true;
                    break;
                }
            }
            if (!flag) {
                System.out.println("no");
            }
        }
        sc.close();
    }
}

File: DSAA/week2/p1.cpp
#include<iostream>
using namespace std;

int n,a[100000];
int T,num;

int main(){
    cin>>n;
    for (int i=0;i<n;i++){
        cin>>a[i];
    }
    cin>>T;
    while (T--){
        cin>>num;
        int l=0,r=n-1;
        while (l<r){
            int mid=(l+r)/2;
            if (a[mid]>=num) r=mid;
            else l=mid+1;
        }
        if (a[l]==num) cout<<"YES"<<endl;
        else cout<<"NO"<<endl;
    }
    return 0;
}